### ISSUE TITLE
Unknown CardService type: MifarePlusStorageSL1 when writing MAD Data to PLUS S chip in SL1

### ISSUE THREAD

**Issue by c3rebro** [islog/liblogicalaccess#207]
It seems using a StorageService on SL1 Plus Cards is not implemented?

```
var cmd = card.getCommands() as MifareCommands;
var cardService = card.getService(LibLogicalAccess.CardServiceType.CST_STORAGE) as StorageCardService;

try
{
     cardService.writeData(mlocation, aiToUse, aiToWrite, new ByteVector(buffer), CardBehavior.CB_AUTOSWITCHAREA);
}
catch (Exception e)
{
...
}
```

The above throws the mentioned LibLogicalAccessNETCE Exception. I tried with:

```
var aiToUse = new MifareAccessInfo
{
		useMAD = _useMADToAuth, (true)
		keyA = mAKeyToUse,
		keyB = mBKeyToUse

		aiToUse.madKeyA = madAKeyToUse;
		aiToUse.madKeyB = madBKeyToUse;
		aiToUse.madGPB = _madGPB; (0xC1)
};
```

**Comment by Maxhy** [comment]
Yeah Mifare Plus are not well supported, see #1. We added such chip support a decade ago but we are not testing it anymore as we used to "not like" this chip for several technical reasons. I guess it could be easily fixed if you are using SL1.

**Comment by c3rebro** [comment]
OK, Thank you for replying to all questions J

 

Von: Maxime C. ***@***.***> 
Gesendet: Freitag, 19. August 2022 10:16
An: islog/liblogicalaccess ***@***.***>
Cc: Steven ***@***.***>; Author ***@***.***>
Betreff: Re: [islog/liblogicalaccess] Unknown CardService type: MifarePlusStorageSL1 when writing MAD Data to PLUS S chip in SL1 (Issue #207)

 

Yeah Mifare Plus are not well supported, see #1 <https://github.com/islog/liblogicalaccess/issues/1> . We added such chip support a decade ago but we are not testing it anymore as we used to "not like" this chip for several technical reasons. I guess it could be easily fixed if you are using SL1.
